diff --git a/scripts/Makefile.package b/scripts/Makefile.package
index 2a90139ecbe1..f3abcc639c46 100644
--- a/scripts/Makefile.package
+++ b/scripts/Makefile.package
@@ -24,7 +24,7 @@ include $(srctree)/scripts/Kbuild.include
 # but the binrpm-pkg target can; for some reason O= gets ignored.
 
 # Remove hyphens since they have special meaning in RPM filenames
-KERNELPATH := kernel-$(subst -,_,$(KERNELRELEASE))
+KERNELPATH := kernel-stoneyridge-$(subst -,_,$(KERNELRELEASE))
 KDEB_SOURCENAME ?= linux-upstream
 KBUILD_PKG_ROOTCMD ?="fakeroot -u"
 export KDEB_SOURCENAME
